src/dataflow/src/source/mod.rs

```diff
@@ -1057,6 +1054,22 @@ where
                                     .unwrap();
                                 partition_metrics.offset_ingested.set(offset.offset);
                                 partition_metrics.messages_ingested.inc();
+
+                                metric_updates.insert(partition, (offset, ts));
+                                messages_until_metric_flush -= 1;
+
+                                // periodically flush metrics in case we're ingesting a large number of messages
+                                // all at once, eg. during initial startup
+                                if messages_until_metric_flush == 0 {
+                                    messages_until_metric_flush = METRIC_FLUSH_INTERVAL;
```

Comment:
Not asking for a change, but curious: normally I see this as:

```rust
messages_until_metric_flush += 1;
// periodically flush metrics in case we're ingesting a large number of messages
// all at once, eg. during initial startup
if messages_until_metric_flush % METRICS_FLUSH_INTERVAL == 0 {
```

are there any pros/cons do doing it either way, in your experience?